lib/services/local/generic/generic.go

```diff
@@ -53,6 +47,8 @@ type ServiceConfig[T Resource] struct {
 	UnmarshalFunc UnmarshalFunc[T]
 }
 
+var _ ServiceCommon[types.User] = (*Service[types.User])(nil)
```

Comment:
I respectfully disagree. I have raised this in the past and there is no consensus nor strong rules about this pattern.

Especially in this particular case, in which we assert that a particular instantiation of a generic type implements the expected interface.

Reply:
This is roughly equivalent to [static assert](https://en.cppreference.com/w/cpp/language/static_assert), which is rather popular.

Reply:
Wildly different languages and paradigms. In Go code usage is king - if your type is used as a certain interface, then it'll be "coerced" to it. If it's not, then implementing that interface is not relevant. Matching signatures are not an accident, but are instead seen as a purposeful act. This all stems from the fact that implementors are not declared but instead matched to the interface after the fact, usually to an interface that is crafted to match it.

Ultimately up to you, I just wanted to give a better explanation.

Reply:
This feels like a partially circular reasoning: you shouldn't do static assertions in Go, because you don't do static assertions in go.

Also, this doesn't really work if the usage is implicit. For example I now had to add this comment:
```
// GetMetadata returns inner metadata from type. Required for dynamically typed helpers like types.GetRevision.
func (w resourceMetadataAdapter[T]) GetMetadata() *headerv1.Metadata {
	return w.resource.GetMetadata()
}
```

If you remove the seemingly unused `GetMetadata()` method, the code compiles fine. Only it won't work correctly, as `types.GetRevision` wont work for such type (other methods too).

Reply:
The comment on GetMetadata (and like methods) is a good call, that would likely be my suggestion.

Reply:
The comment isn't tested or type checked, though.